src/particles/particle_utils.F90

```diff
@@ -77,6 +75,10 @@ subroutine is_part_in_cg(cg, pos, indomain, in, phy, out, fin)
          in  = .true.
          phy = .true.
          out = .true.
+         ijk = ijk_of_particle(pos, dom%edge(:,LO), cg%idl)
+         ijk = max(ijk, cg%l%off)
+         ijk = min(ijk, cg%l%off + cg%l%n_d)
+         fin = cg%leafmap(ijk(xdim), ijk(ydim), ijk(zdim))
```

Comment:
This can lead to reads way outside leafmap bounds when particles start to travel outside the domain

Reply:
fixed by [e32f032](https://github.com/piernik-dev/piernik/pull/532/commits/e32f0328c9844af34e4dc4b2ce3d804c7e829b23)